### PR TITLE
Add model arguments to API and update BERT inference example

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
- The BERT triton inference example is outdated, update to use the latest SDK code for the BERT tokenizer.
- Move commonly defined model arguments in examples to the model server API

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. added --predictor_protocol and --protocol is still kept for backwards compatibility purpose.
2. --model_name, --predictor_host, --predictor_protocol, --predictor_use_ssl are added to the server arguments,
for users who are still defining these arguments in custom model or custom transformer, they can be removed.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
--model_name, --predictor_host, --predictor_protocol, --predictor_use_ssl are now defined in the kserve SDK.
```

### CODE REVIEW

**Review by terrytangyuan — APPROVED** [review]
/lgtm

File: python/kserve/kserve/model.py [hunk 1]
```diff
@@ -59,7 +59,9 @@ def get_latency_ms(start: float, end: float) -> float:
 
 
 class Model:
-    def __init__(self, name: str):
+    def __init__(self, name: str, predictor_host: str = None,
+                 predictor_protocol: str = PredictorProtocol.REST_V1.value,
+                 predictor_use_ssl: bool = False, predictor_request_timeout_seconds: int = 600):
```
Comment (terrytangyuan) [inline python/kserve/kserve/model.py]: How about extracting this into a `predictor_config: PredictorConfig` arg instead?

File: python/kserve/kserve/model_server.py [hunk 2]
```diff
@@ -39,29 +39,46 @@
 DEFAULT_GRPC_PORT = 8081
 
 parser = argparse.ArgumentParser(add_help=False)
+# Model Server Arguments: The arguments are passed to the kserve.ModelServer object
 parser.add_argument("--http_port", default=DEFAULT_HTTP_PORT, type=int,
-                    help="The HTTP Port listened to by the model server.")
+                    help="The HTTP Port listened to by the model server. Default to 8080.")
```
Comment (terrytangyuan) [inline python/kserve/kserve/model_server.py]: The defaults should already been generated in help doc if `default` is specified

Comment (yuzisun) [reply python/kserve/kserve/model_server.py]: I do not see it in the doc, did I miss anything?

```
python3 -m huggingfaceserver --help
usage: __main__.py [-h] [--http_port HTTP_PORT] [--grpc_port GRPC_PORT] [--workers WORKERS] [--max_threads MAX_THREADS] [--max_asyncio_workers MAX_ASYNCIO_WORKERS] [--enable_grpc ENABLE_GRPC] [--enable_docs_url ENABLE_DOCS_URL]
                   [--enable_latency_logging ENABLE_LATENCY_LOGGING] [--configure_logging CONFIGURE_LOGGING] [--log_config_file LOG_CONFIG_FILE] [--access_log_format ACCESS_LOG_FORMAT] [--model_dir MODEL_DIR] [--model_id MODEL_ID]
                   [--enable_streaming ENABLE_STREAMING] [--tensor_parallel_degree TENSOR_PARALLEL_DEGREE] [--max_length MAX_LENGTH] [--do_lower_case DO_LOWER_CASE] [--add_special_tokens ADD_SPECIAL_TOKENS] [--task TASK]

options:
  -h, --help            show this help message and exit
  --http_port HTTP_PORT
                        The HTTP Port listened to by the model server.
  --grpc_port GRPC_PORT
                        The GRPC Port listened to by the model server.
  --workers WORKERS     The number of workers for multi-processing.
  --max_threads MAX_THREADS
                        The number of max processing threads in each worker.
  --max_asyncio_workers MAX_ASYNCIO_WORKERS
                        Max number of asyncio workers to spawn
  --enable_grpc ENABLE_GRPC
                        Enable gRPC for the model server
  --enable_docs_url ENABLE_DOCS_URL
                        Enable docs url '/docs' to display Swagger UI.
  --enable_latency_logging ENABLE_LATENCY_LOGGING
                        Output a log per request with latency metrics.
  --configure_logging CONFIGURE_LOGGING
                        Whether to configure KServe and Uvicorn logging
  --log_config_file LOG_CONFIG_FILE
                        File path containing UvicornServer's log config. Needs to be a yaml or json file.
  --access_log_format ACCESS_LOG_FORMAT
                        Format to set for the access log (provided by asgi-logger).
  --model_dir MODEL_DIR
                        A URI pointer to the model binary
  --model_id MODEL_ID   Huggingface model id
  --enable_streaming ENABLE_STREAMING
                        enable streaming mode
  --tensor_parallel_degree TENSOR_PARALLEL_DEGREE
                        tensor parallel degree
  --max_length MAX_LENGTH
                        max sequence length
  --do_lower_case DO_LOWER_CASE
                        do lower case
  --add_special_tokens ADD_SPECIAL_TOKENS
                        the sequences will be encoded with the special tokens relative to their model
  --task TASK           The task name
```

Comment (yuzisun) [reply python/kserve/kserve/model_server.py]: Need to turn on the option for help formatting with default values.
https://docs.python.org/3/library/argparse.html#formatter-class

Comment (terrytangyuan) [reply python/kserve/kserve/model_server.py]: Yep

Comment (yuzisun) [reply python/kserve/kserve/model_server.py]: Updated

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Pull request merged**